pkg/trace/agent/obfuscate.go

```diff
@@ -176,3 +176,62 @@ func (cco *ccObfuscator) MetaHook(k, v string) (newval string) {
 	}
 	return v
 }
+
+// MetaStructHook checks the tag with the given key and val and returns the final
+// value to be assigned to this tag.
+func (cco *ccObfuscator) MetaStructHook(k string, v []byte) (newval []byte) {
+	switch k {
+	case "appsec":
+		var obfuscated bool
+		var appsec pb.AppSecStruct
```

Comment:
Can we group these?
```suggestion
		var (
			changed bool
			appsec pb.AppSecStruct
		)
```
I'd recommend using `changed`, `remarshal` or `found`. Seem more common names. But I don't feel strongly about this. If you really want to keep yours its fine :)